packages/api/src/mappings/fitbit/body.ts

```diff
@@ -41,7 +46,10 @@ export const mapToBody = (
 
   if (fitbitWeight && fitbitWeight.length > 0) {
     body.weight_samples_kg = fitbitWeight.map(weight => {
-      const dateTime = date + "T" + weight.time;
+      let dateTime = date + "T" + weight.time;
+      dateTime = fitbitUser?.user.timezone
+        ? dayjs.tz(dateTime, fitbitUser?.user.timezone).utc().format()
+        : dateTime;
```

Comment:
since we're using `fitbitUser` to make time calculations, we should probably update [this code](https://github.com/metriport/metriport/blob/04008562a2a3f4bdfd399fab49e97d155a0fd399/packages/api/src/providers/fitbit.ts#L271) to fail if we get the weight without the user - since then the timestamp will be incorrect

Reply:
Yeah, f we need the user for Fitbit ideally we would fail at the entrance, right on the schema validation (in this case, on the data retrieval). This way we can make `fitbitUser` here required, it will simplify the code. :good-news:

Reply:
Makes sense! Fixed!